engine/src/flutter/lib/ui/semantics.dart

```diff
@@ -1111,6 +1111,7 @@ base class _NativeSemanticsUpdateBuilder extends NativeFieldWrapperClass1
       headingLevel >= 0 && headingLevel <= 6,
       'Heading level must be between 1 and 6, or 0 to indicate that this node is not a heading.',
     );
+    print('update node with $role, ${role.index}');
```

Comment:
just for debugging, right?

Reply:
yes, I have been doing some debugging since there is some weirdness in linux engine. For some reason the role enum isn't convert correctly between dart and shell only for linux and windows. I have been using this branch to test in cloudtop since i only have macos gui desktop

Reply:
lol I can't believe these was working for macos https://github.com/flutter/flutter/blob/a336069ba1eb87370c80f9bc250a7c76990b298b/engine/src/flutter/lib/ui/semantics/semantics_update_builder.cc#L124

it changes the property after the value assign to the map. no wonder things go wrong in other platforms